python/src/ecole/core/observation.cpp

```diff
@@ -149,7 +153,69 @@ void bind_submodule(py::module_ const& m) {
 	)");
 	def_before_reset(node_bipartite, "Cache some feature not expected to change during an episode.");
 	def_extract(node_bipartite, "Extract a new :py:class:`NodeBipartiteObs`.");
+    
+    // MILP bipartite observation
+    auto milp_bipartite_obs = py::class_<MilpBipartiteObs>(m, "MilpBipartiteObs", R"(
+		Bipartite graph observation representing the sub-MILP at the latest branch-and-bound node.
+
+		The optimization problem is represented as an heterogenous bipartite graph.
+		On one side, a node is associated with one variable, on the other side a node is
+		associated with one constraint.
+		There exist an edge between a variable and a constraint if the variable exists in the
+		constraint with a non-zero coefficient.
 
+		Each variable and constraint node is associated with a vector of features.
+		Each edge is associated with the coefficient of the variable in the constraint.
+	)");
+	milp_bipartite_obs
+		.def_property_readonly(
+			"variable_features",
+			[](MilpBipartiteObs & self) -> auto& { return self.variable_features; },
+			"A matrix where each row is represents a variable, and each column a feature of "
+			"the variables.")
+		.def_property_readonly(
+			"constraint_features",
+			[](MilpBipartiteObs & self) -> auto& { return self.constraint_features; },
+			"A matrix where each row is represents a constraint, and each column a feature of "
+			"the constraints.")
+		.def_readwrite(
+			"edge_features",
+			&MilpBipartiteObs::edge_features,
+			"The constraint matrix of the optimization problem, with rows for contraints and "
+			"columns for variables.");
+
+	py::enum_<MilpBipartiteObs::VariableFeatures>(milp_bipartite_obs, "VariableFeatures")
+		.value("objective", MilpBipartiteObs::VariableFeatures::objective)
+		.value("is_type_binary", MilpBipartiteObs::VariableFeatures::is_type_binary)
+		.value("is_type_integer", MilpBipartiteObs::VariableFeatures::is_type_integer)
+		.value("is_type_implicit_integer", MilpBipartiteObs::VariableFeatures::is_type_implicit_integer)
+		.value("is_type_continuous", MilpBipartiteObs::VariableFeatures::is_type_continuous)
+		.value("has_lower_bound", MilpBipartiteObs::VariableFeatures::has_lower_bound)
+		.value("has_upper_bound", MilpBipartiteObs::VariableFeatures::has_upper_bound)
+		.value("lower_bound", MilpBipartiteObs::VariableFeatures::lower_bound)
+		.value("upper_bound", MilpBipartiteObs::VariableFeatures::upper_bound);
+
+	py::enum_<MilpBipartiteObs::ConstraintFeatures>(milp_bipartite_obs, "ConstraintFeatures")
+		.value("bias", MilpBipartiteObs::ConstraintFeatures::bias);
+
+	auto milp_bipartite = py::class_<MilpBipartite>(m, "MilpBipartite", R"(
+		Bipartite graph observation function for the sub-MILP at the latest branch-and-bound node.
+
+		This observation function extract structured :py:class:`MilpBipartiteObs`.
+	)");
+	milp_bipartite.def(py::init<bool>(), py::arg("use_normalization") = false, R"(
+		Initialize the logger.
```

Comment:
Logger?

Reply:
I don't know, the same text appears in the constructor of NodeBipartite, I just copy-pasted. Should I change both?

Reply:
Sounds right.